config/routes.rb

```diff
@@ -28,6 +28,12 @@
     resources  :categories, only: :index
     resources  :prospective_users, only: :create
     resources  :assessments, only: [:index, :show, :update, :create] do
+      collection do
+        resources :shared, controller: 'shared_assessments', only: [:show], param: :token do
```

Comment:
I don't understand what you mean with namespacing.

Don't you think it's a little weird to repeat the prefix `/shared/:shared_token/` in three different paths?

Those repeated prefixes seem sinatra-like to me, not rails-like.